presto-main/src/test/java/com/facebook/presto/metadata/TestSignatureBinder.java

```diff
@@ -1034,6 +1035,17 @@ public void testFunction()
                 .withCoercion()
                 .boundTo("integer", new TypeSignatureProvider(paramTypes -> new FunctionType(paramTypes, SMALLINT).getTypeSignature()))
                 .fails();
+
+        // TODO: Support coercion of return type of lambda
+        // Without coercion support for return type of lambda, the return type of function must be `varchar(x)` to avoid need for coercions.
```

Comment:
nit: -> the return type of lambda must be `varchar(x)`. (more clear?)

Does it make sense to add a test case that fails to demonstrate this TODO? (Like some previous test cases), e.g. 

```java
        Signature unboundVarcharApply = functionSignature()
                .returnType(parseTypeSignature("varchar"))
                .argumentTypes(parseTypeSignature("varchar"), parseTypeSignature("function(varchar, varchar)"))
                .build();
        assertThat(unboundVarcharApply)
                .withCoercion()
                .boundTo("varchar", new TypeSignatureProvider(paramTypes -> new FunctionType(paramTypes, createVarcharType(1)).getTypeSignature()))
                .fails();
```